### DISCUSSION TITLE
Static Type Hinting in networkx

### DISCUSSION THREAD

**Discussion by whym1here** [networkx/networkx#5926] [Ideas]
I am not really sure where to start but we can create our custom types using `Type aliases`? 

Advantages: 
- makes the code more readable.
- makes code easier to maintain (depends on the size of the code-base).
- helps the user in identifying the issue with the code which might be difficult in some use cases for dynamically typed languages.

References: 
- https://docs.python.org/3/library/typing.html
- https://numpy.org/doc/stable/reference/typing.html#module-numpy.typing

**Comment by MridulS** [comment]
There is a lot of conversation happening already about this at https://github.com/networkx/networkx/issues/3988, https://github.com/networkx/networkx/pull/4014, https://github.com/networkx/networkx/pull/5886. Let's not create a new discussion post about this :)
